python/create_service_diagnostics_bundle.py

```diff
@@ -138,99 +52,259 @@ def parse_args() -> dict:
+            "cluster list --attached --json", print_output=False)
+
+        if rc != 0:
+            err = "Unexpected error\nstdout: '{}'\nstderr: '{}'".format(
+                stdout, stderr)
+            if "No cluster is attached" in stderr:
+                err = stderr
+            raise ConfigurationException(err, rc)
+
+        try:
+            return json.loads(stdout)
+        except json.JSONDecodeError as e:
+            raise ConfigurationException("Error decoding JSON while getting attached DC/OS cluster: {}".format(e))
+
+    @staticmethod
+    def get_marathon_app(service_name: str):
+        rc, stdout, stderr = sdk_cmd.run_cli(
+            "marathon app show {}".format(service_name), print_output=False
         )
-        log.info("Try '--package-name=%s'", package_name)
-        return (1, {})
-
-    return (
-        0,
-        {
-            "package_name": package_name,
-            "service_name": service_name,
-            "package_version": package_version,
-            "cluster_name": cluster["name"],
-            "bundles_directory": bundles_directory,
-            "dcos_version": cluster["version"],
-            "cluster_url": cluster["url"],
-            "should_prompt_user": should_prompt_user,
-        },
-    )
 
+        if rc == 0:
+            try:
+                return json.loads(stdout)
+            except json.JSONDecodeError as e:
+                err = "Error decoding JSON: {}".format(e)
+        elif "does not exist" in stderr:
+            err = "Service {} does not exist".format(service_name)
+        else:
+            err = "Unexpected error\nstdout: '{}'\nstderr: '{}'".format(
+                stdout, stderr)
+
+        if rc != 0:
+            log.info(
+                "We were unable to get details about '%s'.\nIssue: %s",
+                service_name,
+                stdout,
+            )
+            log.info(
+                "Maybe the '%s' scheduler is not running. That's ok, " +
+                "we can still try to fetch any artifacts related to it",
+                service_name,
+            )
+
+        log.warning(err)
 
-def main(argv) -> int:
-    rc, args = preflight_check()
-    if rc != 0:
-        return rc
+    @staticmethod
+    def validate_package_name(defined_package_name: str, marathon_app, service_name: str):
+        if marathon_app is None:
+            return
+
+        package_name = marathon_app.get("labels", {}).get("DCOS_PACKAGE_NAME")
+
+        if defined_package_name != package_name:
+            err = "Package name given '%s' is different than actual '%s' package name: '%s'".format(
+                defined_package_name,
+                service_name,
+                package_name,
+            )
+            log.info("Try '--package-name=%s'", package_name)
+            raise ConfigurationException(err)
+
+    def check_cluster_state(self):
+        print("Checking attached DC/OS cluster state...")
+
+        clusters = self.clusters
```

Comment:
`self.clusters` is not that much longer than `clusters`, perhaps it does not make much sense to have this intermediate variable?

Reply:
changed